src/core/frontend/framebuffer_layout.cpp

```diff
@@ -479,30 +472,37 @@ FramebufferLayout FrameLayoutFromResolutionScale(u32 res_scale, bool is_secondar
                                      Settings::values.upright_screen.GetValue());
         }
 
-        case Settings::LayoutOption::LargeScreen:
-            if (Settings::values.swap_screen.GetValue()) {
-                width =
-                    (Core::kScreenBottomWidth +
-                     Core::kScreenTopWidth /
-                         static_cast<int>(Settings::values.large_screen_proportion.GetValue())) *
-                    res_scale;
-                height = Core::kScreenBottomHeight * res_scale;
+        case Settings::LayoutOption::LargeScreen: {
+            bool swapped = Settings::values.swap_screen.GetValue();
+            int largeWidth = swapped ? Core::kScreenBottomWidth : Core::kScreenTopWidth;
+            int largeHeight = swapped ? Core::kScreenBottomHeight : Core::kScreenTopHeight;
+            int smallWidth = swapped ? Core::kScreenTopWidth : Core::kScreenBottomWidth;
+            int smallHeight = swapped ? Core::kScreenTopHeight : Core::kScreenBottomHeight;
+            smallWidth =
+                static_cast<int>(smallWidth / Settings::values.large_screen_proportion.GetValue());
+            smallHeight =
+                static_cast<int>(smallHeight / Settings::values.large_screen_proportion.GetValue());
```

Comment:
This code is strange, please initialize the value without it referencing itself.

Reply:
Hmm, if we put this on one line it would be

`const int smallWidth = (int)((swapped ? Core::kScreenTopWidth : Core::kScreenBottomWidth) / Settings::values.large_screen_proportion.GetValue())` (and a similar line for smallHeight) which seems like a lot for a single line. Is that the fix you are imagining, or using a different variable name for the first two lines?

Reply:
Eh, it actually looked okay in code. Went this route, updating now.